scripts/docs.js

```diff
@@ -0,0 +1,144 @@
+/* eslint-disable no-param-reassign */
+/* eslint-disable import/no-extraneous-dependencies */
+
+const typedocs = require('typedocs');
+const fs = require('fs');
+const path = require('path');
+
+const pathToDeclaration = path.resolve(`${__dirname}/../lib/index.d.ts`);
+const pathToREADME = path.resolve(`${__dirname}/../README.md`);
+
+const getType = type => {
+  if (
+    type.startsWith('string') ||
+    type.startsWith('Object') ||
+    type.startsWith('array') ||
+    type.startsWith('boolean')
+  ) {
+    return `\`${type}\``;
+  }
+  if (type.startsWith('{')) {
+    const customType = type.slice(
+      type.indexOf(': ') + 2,
+      type.lastIndexOf(' ')
+    );
+    if (
+      customType.startsWith('string') ||
+      customType.startsWith('Object') ||
+      customType.startsWith('array') ||
+      customType.startsWith('boolean')
+    ) {
+      return `\`${type}\``;
+    }
+    const typeHeader = customType.split('.').pop();
+    const replaced = type.replace(customType, getType(typeHeader));
+    return replaced;
+  }
+  const name = type.split('.').pop();
+  return `[\`${name}\`](#${name})`;
+};
+
+const getPath = obj => {
+  if (obj.parent === null || obj.parent.name === `"${pathToDeclaration}"`)
+    return obj.name;
+  return `${getPath(obj.parent)}.${obj.name}`;
+};
+
+const getMarkdown = (obj, parent = null) => {
+  const { kind } = obj;
+  /* FunctionDeclaration */
+  if (kind === 216) {
+    const { type } = obj;
+    const typeName = type.slice(type.indexOf('<') + 1, type.lastIndexOf('>'));
+    const typeHeading = typeName.split('.').pop();
+    return ''.concat(
+      `### \`${getPath(parent)}.${obj.name}(${obj.parameters
+        .map(param => param.name)
+        .join(', ')})\`\n\n`,
+      `${obj.documentation}\n\n`,
+      `Expects the following parameters:\n\n${obj.parameters.reduce(
+        (acc, val) => acc.concat(`- ${val.name} (${getType(val.type)})\n\n`),
+        ''
+      )}`,
+      `Returns [${type}](#${typeHeading})`,
+      '\n\n'
+    );
+  }
+  /* ClassDeclaration */
+  if (kind === 217) {
+    let md = `## Class <a name="${obj.name}">${obj.name}</a>\n\n`;
+    const constructor = obj.members.shift().parameters[0];
+    md += `Constructed with:\n- ${constructor.name} (\`${constructor.type}\`)\n`;
+    if (obj.members) {
+      obj.members.forEach(child => {
+        md = md.concat(getMarkdown(child));
+      });
+    }
+    return md;
+  }
+  /* InterfaceDeclaration */
+  if (kind === 218) {
+    return ''.concat(
```

Comment:
what do you mean when you say "extra"? I'm just using `''.concat` because it reads so much cleaner than splitting the markdown on either side of the `.concat`.